modules/smartadserverBidAdapter.js

```diff
@@ -1,60 +1,93 @@
+    // to find which one the ad server needs
 
-  return {
-    callBids: function(params) {
-      for (var i = 0; i < params.bids.length; i++) {
-        var bid = params.bids[i];
-        var adCall = url.parse(bid.params.domain);
-        adCall.pathname = '/prebid';
-        adCall.search = {
-          'pbjscbk': '$$PREBID_GLOBAL$$.' + generateCallback(bid),
-          'siteid': bid.params.siteId,
-          'pgid': bid.params.pageId,
-          'fmtid': bid.params.formatId,
-          'ccy': bid.params.currency || 'USD',
-          'bidfloor': bid.params.bidfloor || 0.0,
-          'tgt': encodeURIComponent(bid.params.target || ''),
-          'tag': bid.placementCode,
-          'sizes': bid.sizes.map(size => size[0] + 'x' + size[1]).join(','),
-          'async': 1
-        };
-        adloader.loadScript(url.format(adCall));
-      }
+    // pull requested transaction ID from bidderRequest.bids[].transactionId
+    if (!Array.isArray(validBidRequests)) {
+      validBidRequests = [validBidRequests];
     }
-  };
-};
-
-adaptermanager.registerBidAdapter(new SmartAdServer(), 'smartadserver');
-
-module.exports = SmartAdServer;
+    var bid = validBidRequests[0];
+    const payload = {
+      siteid: bid.params.siteId,
+      pageid: bid.params.pageId,
+      formatid: bid.params.formatId,
+      currencyCode: config.getConfig('currency'),
+      bidfloor: bid.params.bidfloor || 0.0,
+      targeting: bid.params.target && bid.params.target != '' ? bid.params.target : undefined,
+      tagId: bid.adUnitCode,
+      sizes: bid.sizes.map(size => ({
+        w: size[0],
+        h: size[1]
+      })),
+      pageDomain: utils.getTopWindowUrl(),
+      transactionId: bid.transactionId,
+      timeout: config.getConfig('bidderTimeout')
+    };
+    const payloadString = JSON.stringify(payload);
+    return {
+      method: 'POST',
+      url: bid.params.domain + '/prebid/v1',
+      data: payloadString,
+    };
+  },
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function (serverResponse, bidRequest) {
+    const bidResponses = [];
+    if (serverResponse) {
+      const bidResponse = {
+        requestId: bidRequest.bidId,
+        bidderCode: spec.code,
+        cpm: serverResponse.cpm,
+        width: serverResponse.width,
+        height: serverResponse.height,
+        creativeId: serverResponse.creativeId,
+        dealId: serverResponse.dealId,
+        currency: serverResponse.currency,
+        netRevenue: serverResponse.isNetCpm,
+        ttl: serverResponse.ttl,
+        referrer: utils.getTopWindowUrl(),
+        ad: serverResponse.ad
+      };
+      bidResponses.push(bidResponse);
+    }
+    return bidResponses;
+  },
+  getUserSyncs: function (syncOptions) {
+    // iframe || image
+    return undefined;
```

Comment:
If you're not using userSyncs, just don't include this function definition rather than returning `undefined`.